frontend/package.json

```diff
@@ -141,5 +127,21 @@
     "webpack-cli": "^4.10.0",
     "webpack-dev-server": "^4.11.1",
     "webpack-merge": "^5.2.0"
+  },
+  "optionalDependencies": {
```

Comment:
This is a bad design, I'm cannot think of a simple straight up solution for this, is either bring all the build dependencies (webpack, typescript...) to regular dependencies of find a method to install those dependencies **without** installing playwright.

Reply:
We should just remove it from the built image -- we don't need to make them optional.

Reply:
Mmmm but the issue is that CPaaS is blocking the installation of playwright, so if we start to install it, it will block it and kill the build, hence it will fail. I think we need to find a way in order to avoid installing playwright when doing an `npm install` or `npm ci`. 
And then just remove all of the dev dependencies for the runtime image.

Reply:
is there any constraint for which base image can be used as Builder?  e.g customize a base image and push to quay, then use that one for build stages.  1) by pass any whitelist to download binaries 2) speed up build

Reply:
My point is, you can have a node image and install all modules that you need for dashboard, built it into a new image (say, image2) , 
then you use that image2 as your builder, skip "npm install" etc in your dockerfile.

so the download/installation of tools will not be done as part of internal company build process. 
As long as, company accepts your image2 to be pull down.

Reply:
Clarification on my comments here -- spoke with Lucas and we discussed this. I am very much against any long-term / permanent "hack". We talked about `optionalDependencies` and I think what we will do is lean into it. 

Right now the problem is the build system in RHODS doesn't like some of our dependencies. But `optionalDependencies` can be turned off during that build so it omits them and thus should circumvent the issue all together. 

What we will do instead, is treat `optionalDependencies` for all things that are not needed to build or run the product. Anything that is not required for execution of the "primary" product will be optional moving forward.

- `dependencies` are all things included in a production build -- part of our css/js/html deliverables to client's browsers
- `devDependencies` are all things needed to run locally or build for production <sup>1</sup>
- `optionalDependencies` are for all things "extra" -- tests, linting <sup>1</sup>, and any additional tooling for developer support that are not part of the product (like coverage reports or whatever have you)

Note <sup>1</sup> -- eslint may be problematic here -- not sure how much "additional" it is, and how much it is dedicated to being part of the setup for development running. We might be able to disable linting if we do something along the lines of the suggested flow in the NPM Docs ([link](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#optionaldependencies)).

The two states we are expecting today are:

1. Developers will install all dependencies across all 3 categories -- this will give them the full env to do development
2. Production will only install `dependencies` and `devDependencies` and build for production

Eventually we might get to a 3rd state

3. Test runners / CI will run like developers -- installing everything and executing for a non deliverable output (eg just test results / coverage / etc)

Documentation of this effort will need to be done as another ticket. For now, this will serve as the conversation and direction Lucas and I talked about.

Reply:
Documentation will be handled via https://github.com/opendatahub-io/odh-dashboard/issues/1015 -- we do not need to do it part of this PR, other cleanup is needed and this is important to get in asap so we can test it in the full pipeline.